app_dart/lib/src/request_handlers/refresh_github_commits.dart

```diff
@@ -244,4 +275,47 @@ class RefreshGithubCommits extends ApiRequestHandler<Body> {
     throw HttpStatusException(
         HttpStatus.serviceUnavailable, 'GitHub not responding');
   }
+
+  Future<List<String>> _loadBranchRegExps() async {
+    const String path = '/flutter/cocoon/master/dev/branch_regexps.txt';
+    final Uri url = Uri.https('raw.githubusercontent.com', path);
+
+    final HttpClient client = branchHttpClientProvider();
+    try {
+      for (int attempt = 0; attempt < 3; attempt++) {
+        final HttpClientRequest clientRequest = await client.getUrl(url);
+
+        try {
+          final HttpClientResponse clientResponse = await clientRequest.close();
+          final int status = clientResponse.statusCode;
+
+          if (status == HttpStatus.ok) {
+            final String content =
+                await utf8.decoder.bind(clientResponse).join();
+            return content
+                .split('\n')
+                .map((String branch) => branch.trim())
+                .toList();
+          } else {
+            log.warning(
+                'Attempt to download branch_regexps.txt failed (HTTP $status)');
+            return <String>['master'];
+          }
+        } catch (error, stackTrace) {
+          log.error(
+              'Attempt to download branch_regexps.txt failed:\n$error\n$stackTrace');
+        }
+
+        await Future<void>.delayed(gitHubBackoffCalculator(attempt));
+        return <String>['master'];
```

Comment:
This belongs outside the for loop

Reply:
Moved outside. Thank you for pointing this out.